src/test/java/com/example/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,243 @@
+
+import com.example.onlinebookstore.dto.book.BookDto;
+import com.example.onlinebookstore.dto.book.CreateBookRequestDto;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlMergeMode;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Sql(scripts = "classpath:database/categories/add-category.sql")
+@Sql(scripts = {
+        "classpath:database/books/delete-books.sql",
+        "classpath:database/categories/delete-categories.sql"},
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+)
+@SqlMergeMode(SqlMergeMode.MergeMode.MERGE)
+public class BookControllerTest {
+    protected static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @SneakyThrows
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext
+    ) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Add new book into db")
+    void createBook_ValidRequestDto_Success() {
+        CreateBookRequestDto bookRequestDto = new CreateBookRequestDto(
+                "How to cook",
+                "Senior Tomato",
+                "123456789",
+                BigDecimal.valueOf(99),
+                "Very nice book",
+                "image src",
+                Set.of(1L)
+        );
+
+        BookDto expected = new BookDto();
+        expected.setId(1L);
+        expected.setTitle("How to cook");
+        expected.setAuthor("Senior Tomato");
+        expected.setIsbn("123456789");
+        expected.setPrice(BigDecimal.valueOf(99));
+        expected.setDescription("Very nice book");
+        expected.setCoverImage("image src");
+        expected.setCategoriesIds(Set.of(1L));
+
+        String jsonRequest = objectMapper.writeValueAsString(bookRequestDto);
```

Comment:
is it mandatory to convert request to string?

can't you send it as is? pls check